tap_github/scraping.py

```diff
@@ -85,3 +87,67 @@ def _scrape_dependents(url: str, logger: logging.Logger) -> Iterable[Dict[str, A
             time.sleep(1)
         else:
             url = ""
+
+
+def parse_counter(
+    tag: Union[Tag, NavigableString, None], logger: logging.Logger
+) -> int:
+    if not tag:
+        return 0
+    try:
+        title = tag["title"]  # type: ignore
+        if isinstance(title, str):
+            title_string = cast(str, title)
+        else:
+            title_string = cast(str, title[0])
+        return int(title_string.strip())
+    except KeyError:
+        logger.debug(
+            f"Could not parse counter {tag}. Maybe the GitHub page format has changed?"
+        )
+        return 0
+
+
+def scrape_metrics(
+    response: requests.Response, logger: Optional[logging.Logger] = None
+) -> Iterable[Dict[str, Any]]:
+    import re
+
+    from bs4 import BeautifulSoup
+
+    logger = logger or logging.getLogger("scraping")
+
+    soup = BeautifulSoup(response.content, "html.parser")
+
+    try:
+        issues = parse_counter(soup.find("span", id="issues-repo-tab-count"), logger)
+        prs = parse_counter(
+            soup.find("span", id="pull-requests-repo-tab-count"), logger
+        )
+    except IndexError:
+        raise IndexError(
+            "Could not find issues or prs info. Maybe the GitHub page format has changed?"
+        )
+
+    dependents = parse_counter(
+        getattr(soup.find(text=re.compile(" Used by ")), "next_element", None), logger
```

Comment:
Is it worth compiling these regexps at the top of the file so they're cached for all streams, instead of being parsed each time?

Reply:
I checked in the docs, you can also [pass a simple string](https://www.crummy.com/software/BeautifulSoup/bs4/doc/#a-string) to search instead of a regexp. I have no idea if either is faster.

Reply:
Using a string does not seem to work, so let's go with regexps!